### PR TITLE
MAGECLOUD-4530: Create patch for deadlocks related to crons / consumers

### PR DESCRIPTION
**Pull request by BaDos**
### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-4530

### Manual testing scenarios

https://jira.corp.magento.com/browse/MC-6465

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: patches.json [hunk 1]
```diff
@@ -60,6 +60,15 @@
       "2.2.2 - 2.2.3": "MAGECLOUD-1607__overhaul_cron_implementation__2.2.2.patch",
       "2.2.4": "MAGECLOUD-1607__overhaul_cron_implementation__2.2.4.patch"
     },
+    "Fix cron deadlocks and improve cron locking": {
+      "2.2.5 - 2.2.8": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.2.5.patch",
+      "2.2.9": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.2.9.patch",
+      ">=2.2.10 <2.3.0": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.2.10.patch",
+      "2.3.0": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.3.0.patch",
+      "2.3.1": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.3.1.patch",
+      "2.3.2": "MAGECLOUD-4530__fix_cron_deadlocks_and_improve_job_locking__2.3.2.patch",
```
Comment (oshmyheliuk) [inline patches.json]: 2.3.2 has patch version if this patch should be applied to the patch version you need to change constraints

**Comment by BaDos** [comment]
The PR to Magento core was merged: https://github.com/magento/magento2ce/pull/5225

File: patches.json (hunk 1 above)
Comment (BaDos) [reply patches.json]: @oshmyheliuk thanks, fixed.

**Review by shiftedreality — DISMISSED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by NadiyaS** [comment]
QA approved for 2.3.0 and 2.3.3 Magento versions

**Pull request merged**